src/coreclr/jit/lower.cpp

```diff
@@ -6871,20 +6877,18 @@ GenTree* Lowering::LowerConstIntDivOrMod(GenTree* node)
 GenTree* Lowering::LowerSignedDivOrMod(GenTree* node)
 {
     assert((node->OperGet() == GT_DIV) || (node->OperGet() == GT_MOD));
-    GenTree* next = node->gtNext;
 
     if (varTypeIsIntegral(node->TypeGet()))
     {
-        // LowerConstIntDivOrMod will return nullptr if it doesn't transform the node.
-        GenTree* newNode = LowerConstIntDivOrMod(node);
-        if (newNode != nullptr)
+        GenTree* nextNode = nullptr;
+        if (TryLowerConstIntDivOrMod(node, &nextNode))
         {
-            return newNode;
+            return nextNode;
```

Comment:
`assert(nextNode != nullptr)`?

Reply:
The issue is that `nextCode` **can** be a `nullptr` even if the transformation was successful.

But, we could add an assert `assert(nextNode == nullptr);` if the transformation wasn't successful.